### PR TITLE
37: compatibility with NSQ v1.0.0-compact

### PR DESCRIPTION
**Pull request by vic0075**
37: accommodate changes in nsq v1.0.0-compact, also provided backward compatibility

Api call for discovery (/lookup?topic=XXX) has been changed in NSQ v1.0.0-compact (although they've not updated their webpage yet)

`Response in NSQ prior to v1.0.0-compact
{
status_code: 200,
status_txt: "OK",
data: {
channels: [ ],
producers: [
{
remote_address: "127.0.0.1:39310",
hostname: "vikash.tiwari",
broadcast_address: "vikash.tiwari",
tcp_port: 4150,
http_port: 4151,
version: "0.3.8"
}
]
}
}`

`Response in NSQ v1.0.0-compact
{
channels: [
"emailChannel"
],
producers: [
{
remote_address: "127.0.0.1:40258",
hostname: "vikash.tiwari",
broadcast_address: "vikash.tiwari",
tcp_port: 4150,
http_port: 4151,
version: "1.0.0-compat"
}
]
}
`

I have also created the documentation issue [https://github.com/nsqio/nsq/issues/870](https://github.com/nsqio/nsq/issues/870)

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -6,7 +6,7 @@
 
   <groupId>com.github.brainlag</groupId>
   <artifactId>nsq-client</artifactId>
-  <version>1.0.0.RC3</version>
+  <version>1.0.0.RC4</version>
```
Comment (brainlag) [inline pom.xml]: Please don't change the pom.xml unless you add/update a dependency

Comment (vic0075) [inline pom.xml]: Reverted pom.xml in the same branch.

**Comment by vic0075** [comment]
@brainlag: I have reverted the changes from pom.xml and README.md for version thing.
I guess, you guys update it when you release a new version for this.

**Comment by vic0075** [comment]
@brainlag: if possible, can you please create an artifact, I really need it in my project and I don't want to do work around.

**Comment by vic0075** [comment]
@brainlag Gentle Reminder, 
Please create a new artifact version and close the issue: https://github.com/brainlag/JavaNSQClient/issues/37

**Pull request merged**